clients/src/main/java/org/apache/kafka/common/network/TransportLayer.java

```diff
@@ -85,6 +85,13 @@
 
     boolean isMute();
 
+    boolean isInMutableState();
```

Comment:
right now no, this exists as a separate API becuase its a different "aspect". ideally under java8 i could have made this a method with a default impl